index.js

```diff
@@ -537,21 +538,34 @@ class Grid extends react.Component {
         }, SETTINGS_ICON),
         // End of marketplace-header__right
         ),
-            // TODO: Add search bar and sort functionality
-            // react.createElement("div", {
-            //     className: "searchbar--bar__wrapper",
-            // }, react.createElement("input", {
-            //     className: "searchbar-bar",
-            //     type: "text",
-            //     placeholder: "Search for Extensions?",
-            // })),
+        react.createElement("div", {
+            className: "searchbar--bar__wrapper",
+        }, react.createElement("input", {
+            className: "searchbar-bar",
+            type: "text",
+            placeholder: "Search",
+            value: this.state.searchValue,
+            onChange: (event) => this.setState({ searchValue: event.target.value }),
+        })),
         ),
         [ // Add a header and grid for each card type if it has any cards
             { handle: "extension", name: "Extensions" },
             { handle: "theme", name: "Themes" },
             { handle: "snippet", name: "Snippets" },
         ].map((cardType) => {
             const cardsOfType = cardList.filter((card) => card.props.type === cardType.handle)
+                .filter((card) => { // Search filter
+                    const { searchValue } = this.state;
+                    const { title, user } = card.props;
+
+                    if (searchValue.trim() === "")
+                        return card;
+                    else if (
+                        title.toLowerCase().includes(searchValue.trim().toLowerCase()) ||
+                        user.toLowerCase().includes(searchValue.trim().toLowerCase())
+                    )
+                        return card;
```

Comment:
```suggestion
                    if (
                        searchValue.trim() === "" ||
                        title.toLowerCase().includes(searchValue.trim().toLowerCase()) ||
                        user.toLowerCase().includes(searchValue.trim().toLowerCase())
                    ) return card;
```
Can merge together

Reply:
Oh yeah that part came out different on my head lmao, thanks for calling that out.